pkg/operator/sync.go

```diff
@@ -94,6 +94,13 @@ func (optr *Operator) syncClusterAPIController(config *OperatorConfig) error {
 }
 
 func (optr *Operator) syncBaremetalControllers(config *OperatorConfig) error {
+	// First create a Secret needed for the Metal3 deployment
+	err := createMariadbPasswordSecret(optr.kubeClient.CoreV1(), config)
```

Comment:
this could be inlined

Reply:
It's a bit awkward  `syncBaremetalControllers` is now creating a secret. Why is this needed and it wasn't before?

Reply:
Our baremetal installer (https://github.com/openshift-metal3/dev-scripts) was creating the password before. The best place to create it would be right before starting the Metal3 deployment. Once this capability is added to the MAO, then it will be removed from our installer.

Reply:
I see, could you please elaborate and point to where this was created originally in the PR description?

Reply:
Done.

Reply:
Note that dev-scripts is not intended for end-users, so this is a gap which we need to fix to enable non-dev users to interact directly with openshift-install instead.